src/clj/puppetlabs/puppetserver/ringutils.clj

```diff
@@ -85,6 +109,28 @@
         false))
     true))
 
+(schema/defn ^:always-validate request->name :- (schema/maybe schema/Str)
+  "Return the name of the client by parsing the subject of the client cert from
+  the request.  Return nil if the client cert is absent or cannot be parsed."
+  [request :- RingRequest]
+  (let [{:keys [ssl-client-cert]} request]
+    (if ssl-client-cert (get-cert-subject ssl-client-cert))))
+
+(schema/defn ^:always-validate authorized? :- schema/Bool
+  "(SERVER-768) replacement for client-allowed-access?"
+  [rules :- rules/Rules
+   request :- RingRequest]
+  (if-let [name (request->name request)]
+    (let [request-for-tk-authz-schema (merge {:remote-addr ""} request)
```

Comment:
Ring documentation guarantees it will always be present [1].

I ran into this key being absent in our tests which construct mock request objects.  Presumably this means the mock objects aren't to spec, but I didn't go much further than this conclusion as the problematic tests got ripped out anyway.

[1]: https://github.com/mmcgrana/ring/blob/890ffee2e9d57a55a35804745fd1fee9205dc29c/SPEC#L46-L48